keyboards/eggsworks/egg58/info.json

```diff
@@ -0,0 +1,175 @@
+{
+    "keyboard_name": "egg58",
+    "manufacturer": "eggsworks",
+    "url": "https://github.com/eggsworks/egg58",
+    "maintainer": "Travis Mick <travis@eggs.works>",
+    "usb": {
+        "vid": "0xFEED",
+        "pid": "0x3665",
+        "device_version": "2.1.1"
+    },
+    "features": {
+        "bootmagic": true,
+        "rgb_matrix": true
+    },
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "rows": ["C6", "D7", "E6", "B4", "B5"],
+        "cols": ["F6", "F7", "B1", "B3", "B2", "B6"]
+    },
+    "split": {
+        "enabled": true,
+        "transport": {
+            "protocol": "i2c"
+        },
+        "usb_detect": {
+            "enabled": true
+        }
+    },
+    "rgblight": {
+        "pin": "D3",
+        "led_count": 58,
+        "split": true,
+        "split_count": [29, 29],
+        "max_brightness": 160
+    },
```

Comment:
```suggestion
```
Otherwise, looks good to me

Reply:
I'm not super clear on the correct way to set up rgb_matrix using data-driven config - the docs don't seem up to date, I looked at other configs in the repo and it looked like at least the `pin` definition from `rgblight` was still necessary - if that isn't the case, how does the pin get set?

Reply:
`qmk info -kb (name) -f json` will parse the normal config and attempt to render it in json format.

Reply:
Thanks @drashna 

@waffle87 I removed the content of rgblight except for `pin` since that apparently is still necessary